File: memory-common/src/main/java/com/memory/component/jwt/JwtComponent.java
package com.memory.component.jwt;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties(prefix = "jwt.token")
@Getter
@Setter
@ConditionalOnProperty(prefix = "jwt.token", name = "enabled", havingValue = "true")
public class JwtComponent {

    private String secret;
    private Long expiration;
    private Long refresh;

}

File: memory-api/src/main/kotlin/com/memory/component/jwt/JwtComponent.kt
package com.memory.component.jwt

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "jwt.token")
@ConditionalOnProperty(prefix = "jwt.token", name = ["enabled"], havingValue = "true")
class JwtComponent {
    lateinit var secret: String
    var expiration: Long? = null
    var refresh: Long? = null
}